cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java

```diff
@@ -113,10 +117,41 @@ public Partitioning getPartitioning() {
 
   @Override
   public void addPartition(PartitionKey key, String path) {
-    addPartition(key, path, true);
+    addPartition(key, path, new PartitionMetadata(ImmutableMap.<String, String>of()));
   }
 
-  protected void addPartition(PartitionKey key, String path, boolean addToExplore) {
+  @Override
+  public void addPartition(PartitionKey key, String path, PartitionMetadata metadata) {
+    addPartition(key, path, true, metadata);
+  }
+
+  @Override
+  public void updateMetadata(PartitionKey key, PartitionMetadata metadata) {
+    final byte[] rowKey = generateRowKey(key, partitioning);
```

Comment:
Also, why do you feel that it is not a good idea, but it is a good idea in fields of a class?
Wouldn't it be helpful to convey to other readers that this variable won't ever change its reference?